sources/packages/api/src/services/student/student.service.ts

```diff
@@ -176,6 +176,20 @@ export class StudentService extends RecordDataModelService<Student> {
     return this.repo.find({ validSIN: null });
   }
 
+  /**
+   * Gets a student's valid sin status
+   * @returns Students SIN validation status.
+   */
+  async getStudentSinStatus(userId: number): Promise<boolean | null> {
+    const sinStatus = await this.repo
+      .createQueryBuilder("student")
+      .leftJoinAndSelect("student.user", "user")
```

Comment:
A student definitely has to be part of user, so innerJoin is more appropriate here. You can ignore andSelect from join as we do not require any user property here.

Reply:
yes, as the user will be already there, we can use `inner join`

Reply:
actually, I didn't get the other 2 parts `You can ignore andSelect from join as we do not require any user property here.` and `like
.andWhere("user.isActive = :isActive", { isActive: true })
this join might be needed.`

Reply:
@guru-aot thanks for asking this.
Looking at this service, the purpose it solely to get the SIN status, even a inactive user can have a verified SIN status and in that case this would return false for a user who has verified SIN. 

Looking at security aspect you have a point but, I assume if a user is inactive, then that student will not be able to login to our system. In this case there is no scope of showing any banner and blocking access at UI. Looking at the API level where we added this checkSINDecorator, inactive users cannot get past these APIs.

@abschwenker @ann-aot what is your opinion.

Reply:
yes @dheepak-aot , inactive users will not be able to sign in

Reply:
@ann-aot to answer you on the 2nd point what i meant is leftJoin**AndSelect**("student.user", "user") can be replaced with .innerJoin("student.user", "user")